docs/design/networking/01-fix-hostname-usage-related-issues.md

```diff
@@ -0,0 +1,367 @@
+   serialization of those objects. These places will be resolved through
+   the implementation. We need to document the changes in package by
+   package manner. Questions to be answered: What is going to happen to
+   use of `Address` in `com.hazelcast.xxx` package?
+4) To note, we can use a new class to represent our aliases set,
+   `Set<Address>` in our mappings, but I couldn't provide a good
+   abstraction yet.
+
+- Questions about the change:
+    - How does this work in an on-prem deployment? How about on AWS and Kubernetes?
+    - How does the change behave in mixed-version deployments? During a version upgrade? Which migrations are needed?
+    In this effort, we avoid the changes that would break backward compatibility, so we
+    don't expect any backward compatibility issue with respect to this change.
+    - What are the possible interactions with other features or sub-systems inside Hazelcast? How does the behavior of other code change implicitly as a result of the changes outlined in the design document? (Provide examples if relevant.)
+    - Is there other ongoing or recent work that is related? (Cross-reference the relevant design documents.)
+    There is a previously reverted PR: https://github.com/hazelcast/hazelcast/pull/18591 (the related TDD is available in this PR changes), https://github.com/hazelcast/hazelcast/pull/19684
+    - What are the edge cases? What are example uses or inputs that we think are uncommon but are still possible and thus need to be handled? How are these edge cases handled? Provide examples.
+    - Mention alternatives, risks and assumptions. Why is this design the best in the space of possible designs? What other designs have been considered and what is the rationale for not choosing them?
+    - Add links to any similar functionalities by other vendors, similarities and differentiators
+
+
+- Questions about performance:
+    - Does the change impact performance? How?
+    - How is resource usage affected for “large” loads? For example, what do we expect to happen when there are 100000 items/entries? 100000 data structures? 1000000 concurrent operations?
+    - Also investigate the consequences of the proposed change on performance. Pay especially attention to the risk that introducing a possible performance improvement in one area can slow down another area in an unexpected way. Examine all the current "consumers" of the code path you are proposing to change and consider whether the performance of any of them may be negatively impacted by the proposed change. List all these consequences as possible drawbacks.
+        Since we did not perform a lookup from the concurrent
+        `UUID-Set<Address>` and its reverse map in the hot paths, we didn't
+        expect any performance degradation with this change. We perform a simple
+        benchmark on this change and don't see any performance difference with
+        the version before the change. See the benchmark for it:
```

Comment:
to do: please move the benchmark code to https://github.com/hazelcast/internal-benchmarks repository